File: 3-LongestSubstringWithoutRepeatingCharacters/3-LongestSubstringWithoutRepeatingCharacters.py
# Last updated: 13/07/2026, 22:30:51
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        if (not(s)):
            return 0

        state = {s[0] : 0}
        cur_string = s[0]
        result = 1

        for index, char in enumerate(s[1:]):
            if (char in cur_string):
                result = max(result, len(cur_string))
                cur_string = cur_string[cur_string.index(char) + 1:] + char
            else:
                cur_string += char

        return max(result, len(cur_string))

File: 3-LongestSubstringWithoutRepeatingCharacters/3-LongestSubstringWithoutRepeatingCharacters.cpp
// Last updated: 13/07/2026, 22:30:49
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        bitset<256> bset;
        int i = 0 , j = 0, maxLength = 0;
        while(j<s.size()){
            if(bset[int(s[j])] == 1){
                while(s[i] != s[j]){
                    bset[s[i]] = 0;
                    i++;
                }
                i++;
            }
            else{
                bset[int(s[j])] = 1;
            }
            j++;
            if(j-i > maxLength) maxLength = j-i;
        }
        return maxLength;
        
    }
};